config/uint.go

```diff
@@ -2,95 +2,236 @@ package config
 
 import (
 	"fmt"
+	"math"
 	"strconv"
 )
 
-// GetUint returns the uint representation of the value associated with k.
+const (
+	// MaxUint is the maximum value that can be expressed using the uint type.
+	MaxUint = 1<<uintBitSize - 1 // 1<<32 - 1 or 1<<64 - 1
+
+	uintBitSize = 32 << (^uint(0) >> 32 & 1) // 32 or 64
+)
+
+// AsUint returns the int representation of the value associated with k or panics
+// if unable to do so.
+func AsUint(b Bucket, k string) uint {
+	return uint(asUint(b, k, 0, 0, MaxUint))
+}
+
+// AsUintDefault returns the int representation of the value associated with k,
+// or the default value v if k is undefined.
+func AsUintDefault(b Bucket, k string, v uint) uint {
+	return uint(asUintDefault(b, k, 0, uint64(v), 0, MaxUint))
+}
+
+// AsUintBetween returns the int representation of the value associated with k or
+// panics if unable to do so.
+//
+// It panics if the value is not between min and max (inclusive).
+func AsUintBetween(b Bucket, k string, min, max int) uint {
+	return uint(asUint(b, k, 0, uint64(min), uint64(max)))
+}
+
+// AsUintDefaultBetween returns the int representation of the value associated
+// with k, or the default value v if k is undefined.
 //
-// If k is undefined, ok is false and err is nil.
+// It panics if the value is not between min and max (inclusive).
+func AsUintDefaultBetween(b Bucket, k string, v, min, max int) uint {
+	return uint(asUintDefault(b, k, 0, uint64(v), uint64(min), uint64(max)))
+}
+
+// AsUint8 returns the int8 representation of the value associated with k or
```

Comment:
All these specific sized **uint** doc blocks below just say `int#` instead of `uint#`?

Reply:
Fixed.